python/GafferUI/LabelPlugValueWidget.py

```diff
@@ -160,16 +160,20 @@ def _hasUserValue( plug ) :
 	# Sets whether or not the label be rendered in a ValueChanged state.
 	def __setValueChanged( self, valueChanged ) :
 
+		if any( p.direction() == Gaffer.Plug.Direction.Out for p in self.getPlugs() ) :
+			return
+
 		if valueChanged == self.__getValueChanged() :
 			return
 
 		self.__label._qtWidget().setProperty( "gafferValueChanged", GafferUI._Variant.toVariant( valueChanged ) )
 		self.__label._repolish()
+		self.__label._setStyleSheet()
```

Comment:
This will be the only place in Gaffer where we've needed this on anything other than a top-level widget - all other widgets just inherit from their parents. It doesn't seem to be necessary here (testing on Linux), and in fact actually seems to cause clipping of the labels in the ColorSpline UI. What problems were you seeing without it?

Reply:
Without `_setStyleSheet()` I don't get the space reserved for the green dot, so it overlaps with the label.

![image](https://github.com/GafferHQ/gaffer/assets/7873467/3beb3d54-6c3d-448a-8f0a-a4877f438bae)

I found some references online implying that padding would not be taken into account in Qt's `polish()` method. It is a little vague, but if the [Qt documentation](https://doc.qt.io/qt-5/qstyle.html#polish)'s mention of not changing geometry includes padding, then that sounds like intended behavior.

I don't seem to have any clipping of labels (on Windows) :
![image](https://github.com/GafferHQ/gaffer/assets/7873467/11c3a1e0-ace4-4c94-bfb2-66cbc645da56)

Reply:
Resolved in https://github.com/GafferHQ/gaffer/pull/5737/commits/b25b038edbb40d12dfcd42a143a8f24d26e055c5 and https://github.com/GafferHQ/gaffer/pull/5737/commits/d3aaa2f4146b4b6c848bd0ca02c368db6e1d62e4 where I added metadata to control whether or not the icon is shown, and hiding it for spline plugs.